src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs

```diff
@@ -25798,6 +25798,70 @@ public void Test1(IDo<object> impl)
             Assert.Equal("IResult<System.Int32>", doSymbol.ReturnType.ToTestDisplayString());
         }
 
+        [Fact]
+        [WorkItem(40430, "https://github.com/dotnet/roslyn/issues/40430")]
+        public void MissingTypeArgumentInBase_ValueTuple()
```

Comment:
>MissingTypeArgumentInBase_ValueTuple [](start = 20, length = 36)

Consider adding a test that involves nullable annotations on tuple elements.

Reply:
Perhaps, dynamic as well

---
In reply to: [362911136](https://github.com/dotnet/roslyn/pull/40704#discussion_r362911136) [](ancestors = 362911136)

Reply:
Looking at the decoder for nullable and dynamic, they do not have the same logic (catching exception then falling back based on presence of use-site errors).

Reply:
> Looking at the decoder for nullable and dynamic, they do not have the same logic (catching exception then falling back based on presence of use-site errors).

I think we should still add a test, if possible.

---
In reply to: [363600527](https://github.com/dotnet/roslyn/pull/40704#discussion_r363600527) [](ancestors = 363600527)

Reply:
I added a test for nullable decoding.
But I didn't add a test for dynamic. It's more trouble to test because the type is part of corlib, and since we don't think there is a bug, it doesn't feel like the extra effort is worthwhile. Sounds okay?

---
In reply to: [363892777](https://github.com/dotnet/roslyn/pull/40704#discussion_r363892777) [](ancestors = 363892777,363600527)